core/src/main/java/org/infinispan/protostream/impl/TagWriterImpl.java

```diff
@@ -309,6 +309,8 @@ public ProtoStreamWriterImpl getProtoStreamWriter() {
 
    //todo [anistor] need to provide a safety mechanism to limit message size in bytes and message nesting depth on write ops
    private abstract static class Encoder {
+
+      protected final StringUtil.Utf8Helper utf8Helper = StringUtil.getUtf8Helper();
```

Comment:
We need to make this static, so the if block below can surely be optimized away as necessary.

Reply:
static and final :pray: